src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/DirectorManager.java

```diff
@@ -47,4 +56,21 @@ public Player getDirector(@NotNull Craft craft) {
         return director;
     }
 
+    protected Craft getDirectingCraft(Metadatable directed) {
+        if(!Config.EnableTNTTracking)
+            return null;
+
+        List<MetadataValue> meta = directed.getMetadata("MCC-Sender");
+        if(meta.isEmpty())
+            return null;
+
+        Player sender = Bukkit.getPlayer(UUID.fromString(meta.get(0).asString()));
+        if (sender == null || !sender.isOnline())
+            return null;
+
+        Craft c = CraftManager.getInstance().getCraftByPlayer(sender);
+        if (c == null || c.getSinking())
+            return null;
+        return c;
+    }
```

Comment:
This won't work correctly for fireballs.

Reply:
How so?

Reply:
It refers to TNT tracking config, so if that is enabled but fireball tracking is not it will throw an NPE.